lib/rubocop/cop/rspec/example_without_description.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for examples without a description.
+      #
+      # RSpec allows for auto-generated example descriptions when there is no
+      # description provided or the description is an empty one.
+      #
+      # This cop removes empty descriptions.
+      # It also defines whether auto-generated description is allowed, based
+      # on the configured style.
+      #
+      # This cop can be configured using the `EnforcedStyle` option
+      #
+      # @example `EnforcedStyle: always_allow`
+      #   # bad
+      #   it('') { is_expected.to be_good }
+      #   it '' do
+      #     result = service.call
+      #     expect(result).to be(true)
+      #   end
+      #
+      #   it { is_expected.to be_good }
+      #   it do
+      #     result = service.call
+      #     expect(result).to be(true)
+      #   end
+      #
+      # @example `EnforcedStyle: single_line_only`
+      #   # bad
+      #   it('') { is_expected.to be_good }
+      #   it do
+      #     result = service.call
+      #     expect(result).to be(true)
+      #   end
+      #
+      #   good
+      #   it { is_expected.to be_good }
+      #
+      # @example `EnforcedStyle: disallow`
+      #   # bad
+      #   it { is_expected.to be_good }
+      #   it do
+      #     result = service.call
+      #     expect(result).to be(true)
+      #   end
+      class ExampleWithoutDescription < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG_DEFAULT_ARGUMENT = 'Omit the argument when you want to ' \
+                               'have auto-generated description.'.freeze
+        MSG_ADD_DESCRIPTION  = 'Add a description.'.freeze
+
+        def_node_matcher :example?, Examples::ALL.send_pattern
+        def_node_matcher :example_description, '(send nil? _ $(str $_))'
+
+        def on_send(node)
+          return unless example?(node)
+
+          check_example_without_description(node)
+
+          example_description(node) do |message_node, message|
+            return unless message.to_s.empty?
+
+            add_offense(message_node, message: MSG_DEFAULT_ARGUMENT)
+          end
+        end
+
+        private
+
+        def check_example_without_description(node)
+          return unless dissalow_empty_description?(node)
+
+          _send, _method, arg = *node
+          add_offense(node, message: MSG_ADD_DESCRIPTION) if arg.nil?
+        end
+
+        def dissalow_empty_description?(node)
+          return true if style == :single_line_only && node.parent.multiline?
+
+          style == :disallow
+        end
```

Comment:
I’m always trying to come up with alternatives to `return <boolean logic>\n<other boolean logic>`. How about this:
```rb
def disallow_empty_description?(node)
  style == :disallow ||
    (style == :single_line_only && node.parent.multiline?)
end
```

Reply:
I'm not fan either, but I also dislike mixing || and &&. Anyway, I'll use your version, as it looks cleaner indeed